### DISCUSSION TITLE
Find a contact plan and creat a geometry

### DISCUSSION THREAD

**Discussion by FerZanoni** [IfcOpenShell/IfcOpenShell#2830] [Q&A]
Hi everyone,

I want to know it's possible to find which elements share the same planne (like a wall leaning against a pillar) and create a object with a 1mm of thickness for this plane. 

The closest I got was with this writing that returns the name of the elements that have contact, I want it to return the GUID of the elements and in the contact plane of the elements create an element in the same dimensions of the contact plane with 1mm thick. In this code I put Wall to find, but I want to every element.

import ifcopenshell

ifc_file = ifcopenshell.open("**PUT THE IFC FILE PATH HERE**")

elements = ifc_file.by_type("IfcElement")

for element1 in elements:
    for element2 in elements:
        if element1.id() == element2.id():
            continue

        if (
            element1.is_a("IfcWall") and element2.is_a("IfcWall") and
            element1.ObjectPlacement.PlacementRelTo.id() == element2.ObjectPlacement.PlacementRelTo.id()
        ):
            print(f"Os elementos {element1.Name} e {element2.Name} compartilham o mesmo plano de referência")

            new_element = ifc_file.create_entity(
                "IfcWallStandardCase",
                Name="Novo elemento",
                ObjectPlacement=element1.ObjectPlacement
            )

            ifc_file.add(new_element)
            ifc_file.write("**PUT THE PATH HERE TO SAVE**")

**Comment by Moult** [comment]
Can you clarify what you're after? Are you trying to find any pairs of elements which are touching, but not intersecting, and create planes that correlate with the portions of contacting surfaces?

That sounds pretty tricky. ifcClash can show you how to use the geom tree to do broadphase collision of objects as a start.

**Reply by aothms** [reply]
If you replicate the functionality of ifcopenshell.geom.tree in (python)occ and use BRepDistShapeShape you get the minimal distance between shapes, but also the point on each face (or edge/vertex) that realizes this distance. You can then project the face onto the other and do a boolean intersection of both. Extrude that. It can work.

Alternatively we did an investigation of this approach https://github.com/tudelft3d/IfcOpenShell/blob/cgal/src/ifcconvert/validate_space_boundaries.cpp dilating every element and doing 3d intersection, then flattening to face.

So all in all we don't have a readily made solution available, but there's starting points you can use as a basis.
